### PR TITLE
[docs] guideline for how to enable new third-party library

### PR DESCRIPTION
**Pull request by mkrainiuk**
This is guideline for how to integrate new third-party library to oneMKL interfaces with simple examples and helper python scripts

Rendered guideline you can review in my fork: https://github.com/mkrainiuk/oneMKL/blob/master/docs/create_new_backend.rst

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # Contributing Guidelines
 If you have improvements, new libraries integrated under oneAPI Math Kernel Library (oneMKL) interfaces, or new interfaces to contribute to the oneMKL Specification, please send us your pull requests! For getting started, see GitHub [howto](https://help.github.com/en/articles/about-pull-requests).
 
-The current guidelines are a work in progress.
+For how to enable new third-party library see the [guideline](docs/create_new_backend.rst)
```
Comment (petercad) [inline CONTRIBUTING.md:4]: guideline -> guidelines; also a period at the end!

File: docs/create_new_backend.rst [hunk 2]
```diff
@@ -0,0 +1,483 @@
...
+        asum_postcondition(queue, n, x, incx, result);
+    }
+
+
+.. _integrate_header_files:
+
+2. Integrate Header Files
+-------------------------
+
+Below you can see oneMKL Header-based Part Structure:
```
Comment (petercad) [inline docs/create_new_backend.rst:86]: "Below you can see the oneMKL directory structure for headers:" might be easier to understand.

File: docs/create_new_backend.rst [hunk 3]
```diff
@@ -0,0 +1,483 @@
...
+        onemkl/
+            onemkl.hpp -> oneMKL spec APIs
+            types.hpp  -> oneMKL spec types
+            detail/    -> implementation specific header files
+                exceptions.hpp        -> oneMKL exception classes
+                libraries.hpp         -> list of oneMKL libraries
+                backends.hpp          -> list of oneMKL backends
+                backends_selector.hpp -> oneMKL runtime dispatcher based on queue
+            blas/
+                predicates.hpp -> oneMKL BLAS pre-√ post-√
```
Comment (petercad) [inline docs/create_new_backend.rst:100]: I'd write out "pre-check", "post-check", etc.

File: CONTRIBUTING.md (hunk 1 above)
Comment (mkrainiuk) [reply CONTRIBUTING.md:4]: Thanks! I will update it

File: docs/create_new_backend.rst (hunk 2 above)
Comment (mkrainiuk) [reply docs/create_new_backend.rst:86]: We also have headers in src/, so I tried to specify that this is structure inside include/. Do you think it's not really important?

File: docs/create_new_backend.rst (hunk 3 above)
Comment (mkrainiuk) [reply docs/create_new_backend.rst:100]: Agree, I will update it

File: docs/create_new_backend.rst (hunk 2 above)
Comment (petercad) [reply docs/create_new_backend.rst:86]: Ah, right :) Maybe "The diagram below shows the directory structure of the oneMKL external headers"?

Comment (mkrainiuk) [reply docs/create_new_backend.rst:86]: Sound better, I guess "detail" folders are not really external, so how about "Below you can see structure of oneMKL top-level include directory"?

Comment (petercad) [reply docs/create_new_backend.rst:86]: That sounds good to me, too.

Comment (mkrainiuk) [reply docs/create_new_backend.rst]: Thank you! I've pushed all suggested changes

**Pull request merged**